### PR TITLE
FEATURE: Added API Endpoints for individual Update health and cow milking jobs 

### LINKED ISSUE #44: FEATURE: Create backend endpoints for new milking jobs
## Overview
In this issue, we are going to implement the new backend endpoints necessary to implement issue #11 

## Acceptance
- [x] Either: the backend endpoint that launches the job to milk the cows can take an **optional ** parameter specifying a particular commonsId (if not specified, then the job is run for all commons).  OR there is a separate endpoint for milking cows for a particular commons.
- [x] Either: the backend endpoint that launches the job to update cow health can take an **optional ** parameter specifying a particular commonsId (if not specified, then the job is run for all commons).  OR there is a separate endpoint for updating cow health for a particular commons.

### PR DESCRIPTION
**Pull request by MPriston**
## Overview

In this new and revised PR, we introduce the changes requested on PR #81 and introduce new API endpoints that will allow the admin to specify a common when running the milking of the cows and updating cow health jobs

## DEPENDENCIES
In order for this PR to be merged PR #91 needs to be merged first!!  (Update... it was merged)

## Screenshots (Optional)

image
image
image
image

## Validation (Optional)

1. Download this branch.
2. Run the project.
3. Go to this swagger.
4. Test new API Endpoints.

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 

## Linked Issues

Closes #44

### CODE REVIEW

**Comment by MPriston** [comment]
Decided to create a new PR from the previous #81

File: src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJob.java [hunk 1]
```diff
@@ -27,7 +27,7 @@ public class MilkTheCowsJob implements JobContextConsumer {
     @Getter
     private ProfitRepository profitRepository;
 
-    public String formatDollars(double amount) {
```
Comment (publicqi) [inline src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJob.java:30]: Why static?

**Review by publicqi — DISMISSED** [review]
The code is generally good. Maybe we can decouple the code by implementing the two "Ind" classes to be child class of the non-"Ind" ones.

I don't spot any bug here.

File: src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJob.java (hunk 1 above)
Comment (MPriston) [reply src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJob.java:30]: Só that It can be accessed by the other static method inside the class

**Review by dscpsyl — APPROVED** [review]
Thanks for the changes!

**Review by GDW1 — APPROVED** [review]

**Pull request merged**